student-os/scripts/materials_convert.py

```
@@ -461,6 +468,151 @@ def get_mineru_client(ctx: ConversionContext) -> Any:
+            response.read()
+    except urllib.error.HTTPError as exc:
+        detail = exc.read().decode("utf-8", errors="replace").strip()
+        raise RuntimeError(f"MinerU v1 Agent upload HTTP {exc.code}: {detail}") from exc
+    except urllib.error.URLError as exc:
+        raise RuntimeError(f"MinerU v1 Agent upload failed: {exc.reason}") from exc
+
+
+def http_text(url: str, *, timeout: int = 300) -> str:
+    url = mineru_agent_url_allowed(url)
+    try:
+        with urllib.request.urlopen(url, timeout=timeout) as response:
```

Ruff (0.16.1)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)